mm/2s2h/Enhancements/Masks/EasyMaskEquip.cpp

```diff
@@ -0,0 +1,337 @@
+        gMaskEquipState = MASK_EQUIP_PENDING_EQUIP;
+        Audio_PlaySfx(NA_SE_SY_DECIDE);
+    }
+    // Process the B button: cancel any pending operation.
+    else if (CHECK_BTN_ALL(pressedButtons, BTN_B)) {
+        gPendingMask = ITEM_NONE;
+        gMaskEquipState = MASK_EQUIP_NONE;
+        gLastEquippedMask = ITEM_NONE;
+    }
+}
+
+// Hook Registrations
+
+void RegisterEasyMaskEquip() {
+    // Process transformation mask input during pause menu updates.
+    COND_HOOK(OnKaleidoUpdate, CVAR, [](PauseContext* pauseCtx) {
+        if (pauseCtx->pageIndex == PAUSE_MASK)
+            handleEasyMaskEquip(pauseCtx);
+    });
+
+    // Hook for drawing mask icons.
+    COND_VB_SHOULD(VB_DRAW_MASK_ITEM, CVAR, {
+        u16* itemId = va_arg(args, u16*);
+        s16* index = va_arg(args, s16*);
+        if (isTransformationMask(static_cast<ItemId>(*itemId))) {
+            renderMaskItem(&gPlayState->pauseCtx, *itemId, *index);
+            *should = false;
+        } else {
+            *should = true;
+        }
+    });
+
+    // Hook for drawing the equip border (transformation masks only).
+    COND_ID_HOOK(BeforeKaleidoDrawPage, PAUSE_MASK, CVAR, [](PauseContext* pauseCtx, u16) {
+        if (pauseCtx->pageIndex == PAUSE_MASK) {
+            allocateEquipBorderVertices(gPlayState->state.gfxCtx);
+            updateEquipBorderVertices(pauseCtx);
+            drawEquipBorder(pauseCtx);
+        }
+    });
+
+    // Hook for equipping pending transformation masks when the pause menu closes.
+    COND_HOOK(OnKaleidoClose, CVAR, [](u16 pauseIndex) {
+        if (pauseIndex != PAUSE_MASK)
+            return;
+
+        Player* player = GET_PLAYER(gPlayState);
+        if (gMaskEquipState == MASK_EQUIP_PENDING_UNEQUIP) {
+            if (Player_GetCurMaskItemId(gPlayState) != ITEM_NONE)
+                Player_UseItem(gPlayState, player, static_cast<ItemId>(Player_GetCurMaskItemId(gPlayState)));
+            gLastEquippedMask = ITEM_NONE;
+            gMaskEquipState = MASK_EQUIP_NONE;
+        } else if (gMaskEquipState == MASK_EQUIP_PENDING_EQUIP) {
+            Player_UseItem(gPlayState, player, gPendingMask);
+            gLastEquippedMask = gPendingMask;
+            gPendingMask = ITEM_NONE;
+            gMaskEquipState = MASK_EQUIP_TRANSFORMING;
+        }
+    });
+
+    // Hook for updating equip state during actor updates.
+    COND_HOOK(OnActorUpdate, CVAR, [](Actor* actor) {
+        if (actor->id != ACTOR_PLAYER)
+            return;
+        Player* player = reinterpret_cast<Player*>(actor);
+        if (gMaskEquipState == MASK_EQUIP_TRANSFORMING && player->transformation == PLAYER_FORM_HUMAN)
+            gMaskEquipState = MASK_EQUIP_NONE;
+    });
+
+    // Hook for controlling the display of the selected item textbox.
+    COND_VB_SHOULD(VB_KALEIDO_DISPLAY_ITEM_TEXT, CVAR, {
+        *should = !isTransformationMask(static_cast<ItemId>(gPlayState->pauseCtx.cursorItem[PAUSE_MASK]));
```

Comment:
This appears to be causing an issue with persistent bunny hood, causing it to show the text when toggling it.

Instead I think you want
```suggestion
        if (isTransformationMask(static_cast<ItemId>(gPlayState->pauseCtx.cursorItem[PAUSE_MASK]))) {
          *should = false;
        }
```